### ISSUE TITLE
test_backend.pgf.check_for(texsystem) does not do what it says...

### ISSUE THREAD

**Issue by JanSchulz** [matplotlib/matplotlib#5921]
```
def check_for(texsystem): #<- different commands are asked for, but not actually checked below...
    header = """
    \\documentclass{minimal}
    \\usepackage{pgf}
    \\begin{document}
    \\typeout{pgfversion=\\pgfversion}
    \\makeatletter
    \\@@end
    """
    try:
        latex = subprocess.Popen(["xelatex", "-halt-on-error"], #<- why not texsystem? 
                                 stdin=subprocess.PIPE,
                                 stdout=subprocess.PIPE)
        stdout, stderr = latex.communicate(header.encode("utf8"))
    except OSError:
        return False

    return latex.returncode == 0
```

Not sure if thats on purpose, but if so it should be documented...

**Comment by jenshnielsen** [comment]
I think this is a bug. I have fixed it as part of #5727 which still waits for Travis to approve cm-super before it can be merged.

**Comment by jenshnielsen** [comment]
@pwuertz  does the change in 7d67737 make sense to you?

**Comment by pwuertz** [comment]
Right, this is a bug/typo in `check_for()`

**Comment by jenshnielsen** [comment]
fixed in #5727
